mitodl/open-discussions
# Profile ImageFields

#### What are the relevant tickets?
Closes #693

#### What's this PR do?
- Adds `image_file`, `image_small_file`, `image_medium_file` ImageFields.  
- Adds a utility function to return the most appropriate image for a profile (for example, `image_small_file.url` if not `None`, else `image_small` if not None, else the default profile image.
- Adds support for enabling S3 media storage 

#### How should this be manually tested?
- Log in to OD with an existing user that has a profile image.  The image should still display.
- From Micromasters, create a new user with a profile image and go to OD.  The profile image should display.
- Log in to OD with a new or existing user that has no profile image.  A bland gray default profile image should display.
- Use the API to update the `image_small_file` field of your profile:
  - Get a `jwt_token`:
    ```python
    from fixtures.users import staff_jwt_token
    from django.contrib.auth.models import User
    user = User.objects.get(username=<your_username>)
    staff_jwt_token(None, user)
    ```
  - Send an API PATCH request to update your profile image:
    ```
    curl --request PATCH -H "Authorization: Bearer  <jwt_token>" -F 'image_small_file=@<image_name.jpg>'  http://od.odl.local:8063/api/v0/profiles/<username>/
    ```
  - Reload OD, you should see the new profile image.
